cache/src/main/java/net/runelite/cache/io/OutputStream.java

```diff
@@ -181,16 +182,17 @@ public void writeLengthFromMark(int var1)
 
 	public void writeString(String str)
 	{
-		byte[] b;
-		try
-		{
-			b = str.getBytes("ISO-8859-1");
-		}
-		catch (UnsupportedEncodingException ex)
-		{
-			throw new RuntimeException(ex);
-		}
-		writeBytes(b);
+		Charset utf8charset = Charset.forName("UTF-8");
+		Charset cp1252charset = Charset.forName("Cp1252");
+
+		ByteBuffer inputBuffer = ByteBuffer.wrap(str.getBytes());
+
+		CharBuffer data = utf8charset.decode(inputBuffer);
```

Comment:
Why are you converting from utf16 to utf 8 to utf 16 to cp1252? Just go from utf16 to cp1252.

Reply:
Is that not what adam suggested and what I tried with the snippet 2 comments back or do you have another way in mind? the snippet above didn't have work for me.

Reply:
It is what Adam suggested; you should figure out why it does not work because what you have submitted looks very wrong and you don't seem to understand why it works.

Reply:
The following patch, while following your recommendation of encoding bytes directly as utf-16 -> cp1252, doesn't seem to fix it for me.

```diff
diff --git a/cache/src/main/java/net/runelite/cache/io/OutputStream.java b/cache/src/main/java/net/runelite/cache/io/OutputStream.java
index 77853598b..c1591bbdb 100644
--- a/cache/src/main/java/net/runelite/cache/io/OutputStream.java
+++ b/cache/src/main/java/net/runelite/cache/io/OutputStream.java
@@ -27,7 +27,7 @@ package net.runelite.cache.io;
 import com.google.common.base.Preconditions;
 import java.io.IOException;
 import java.nio.ByteBuffer;
-import java.nio.charset.StandardCharsets;
+import java.nio.charset.Charset;
 
 public final class OutputStream extends java.io.OutputStream
 {
@@ -181,7 +181,7 @@ public final class OutputStream extends java.io.OutputStream
 
 	public void writeString(String str)
 	{
-		writeBytes(str.getBytes(StandardCharsets.ISO_8859_1));
+		writeBytes(str.getBytes(Charset.forName("Cp1252")));
 		writeByte(0);
 	}
```